Sources/NIO/BaseSocketChannel.swift

```diff
@@ -755,9 +751,36 @@ class BaseSocketChannel<T: BaseSocket>: SelectableChannel, ChannelCore {
         }
     }
 
+    final func readEOF() {
+        if self.lifecycleManager.isRegistered {
+            self.safeReregister(interested: self.interestedEvent.subtracting(.readEOF))
+
+            while self.lifecycleManager.isActive {
+                var didReceiveEOF: Bool = false
+                self.readable0(didReceiveEOF: &didReceiveEOF)
```

Comment:
Why does this use an inout instead of a return value?

Reply:
@Lukasa/@normanmaurer because of the 'blind boolean' problem. We don't have named return values and I didn't really wanted to introduce a new enum just for that. Happy to change to whatever you both think is best.

Reply:
I think adding an enum would be ok.

Reply:
I'm perfectly fine with adding an enum for this. :smile: